### PR TITLE
bug: can't start websocket

### PR DESCRIPTION
**Pull request by dagnis-g**

### CODE REVIEW

**Comment by dagnis-g** [comment]
[trading.log](https://github.com/dagnis-g/okx/files/9132100/trading.log)

File: src/main/java/com/example/tradingapp/websocket/MyWebsocketClient.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.web.socket.WebSocketHttpHeaders;
+import org.springframework.web.socket.client.WebSocketClient;
+import org.springframework.web.socket.client.standard.StandardWebSocketClient;
+
+import java.net.URI;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class MyWebsocketClient {
```
Comment (azheltishev) [inline src/main/java/com/example/tradingapp/websocket/MyWebsocketClient.java:15]: let's come up with a more fitting class name

File: src/main/java/com/example/tradingapp/websocket/MyWebsocketClient.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    private final MyWebsocketHandler myWebsocketHandler;
+    private static String URL = "wss://wspap.okx.com:8443/ws/v5/private?brokerId=9999";
```
Comment (azheltishev) [inline src/main/java/com/example/tradingapp/websocket/MyWebsocketClient.java:18]: this URL would be different for demo/prod environments and can be moved to properties

File: src/main/java/com/example/tradingapp/websocket/MyWebsocketHandler.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                List<SubscribeArg> args = new ArrayList<>();
+                args.add(subscribeArg);
+                var subscribeRequest = SubscribeRequest.builder()
+                        .op("subscribe")
+                        .args(args)
+                        .build();
+                String subscribeJson = mapper.writeValueAsString(subscribeRequest);
+                session.sendMessage(new TextMessage(subscribeJson));
+            } else {
+                log.warn("Login not successful");
```
Comment (azheltishev) [inline src/main/java/com/example/tradingapp/websocket/MyWebsocketHandler.java:57]: we can print `jsonNode` here, to see why login wasn't successful

File: src/main/java/com/example/tradingapp/websocket/MyWebsocketHandler.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                        .channel("orders")
+                        .build();
+                List<SubscribeArg> args = new ArrayList<>();
+                args.add(subscribeArg);
+                var subscribeRequest = SubscribeRequest.builder()
+                        .op("subscribe")
+                        .args(args)
+                        .build();
+                String subscribeJson = mapper.writeValueAsString(subscribeRequest);
+                session.sendMessage(new TextMessage(subscribeJson));
```
Comment (azheltishev) [inline src/main/java/com/example/tradingapp/websocket/MyWebsocketHandler.java:55]: just a minor style concern, but we can try extracting this code to a different method and calling it here

File: src/main/java/com/example/tradingapp/websocket/MyWebsocketHandler.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.web.socket.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+@Slf4j
+public class MyWebsocketHandler implements WebSocketHandler {
```
Comment (azheltishev) [inline src/main/java/com/example/tradingapp/websocket/MyWebsocketHandler.java:19]: same here, let's use different class name

**Review by azheltishev — APPROVED** [review]

**Pull request merged**